masonry/src/contexts.rs

```diff
@@ -577,9 +583,27 @@ impl_context_method!(
 pub struct TimerToken;
 
 impl EventCtx<'_> {
-    // TODO - Document
+    // TODO - clearly document all semantics of pointer capture when they've been decided on
     // TODO - Figure out cases where widget should be notified of pointer capture
     // loss
+    /// Capture the pointer in the current widget.
+    ///
+    /// Pointer capture is only allowed during a [`PointerDown`] event. It is a logic error to
+    /// capture the pointer during any other event.
+    ///
+    /// A widget normally only receives pointer events when the pointer is inside the widget's
+    /// layout box. When the pointer is captured by a widget, that widget will continue receiving
+    /// pointer events when the pointer is outside the widget's layout box.
+    ///
+    /// The pointer cannot be captured by multiple widgets at the same time. If a widget has
+    /// captured the pointer and another widget captures it, the first widget loses the pointer
+    /// capture.
+    ///
+    /// Discussion: there is currently some confusion about whether a widget can be considered hot
+    /// when some other widget has captured the pointer (for example, when clicking one widget and
+    /// dragging to the next). The documentation should clearly state the resolution.
```

Comment:
I don't think there's any confusion, but maybe I missed a Zulip thread.

The semantics are this: if a widget has pointer capture, either that widget is hot/hovered, or no widget is. This matches web behavior.

Reply:
I duplicated the discussion point from here

https://github.com/linebender/xilem/blob/6c4951635da34cf04632332114e7b04ca343d8c0/masonry/src/contexts.rs#L254-L260

I probably should've referenced it instead of copying. If the question is resolved I can remove the comment from this PR, and create a new PR removing the old discussion point on is_hot and document that behavior.

Somewhat related to this discussion is this Zulip thread about pointer exclusivity: https://xi.zulipchat.com/#narrow/stream/354396-xilem/topic/Exclusive.20events.20when.20interacting

Reply:
> If the question is resolved I can remove the comment from this PR, and create a new PR removing the old discussion point on is_hot and document that behavior.

Sure.

> Somewhat related to this discussion is this Zulip thread about pointer exclusivity: https://xi.zulipchat.com/#narrow/stream/354396-xilem/topic/Exclusive.20events.20when.20interacting

Oh, I forgot about that thread. I wonder if we still have an active issue somewhere referencing it. We might want to add a comment with a link to either a github issue or the zulip thread.

Reply:
I've removed the discussion point. I can update the docs with the behavior of capture/hot on top of this PR. Easiest after it's merged.